### ISSUE TITLE
can't load blockchain 0.9.14

### ISSUE THREAD

**Issue by mega2k0** [aeonix/aeon-legacy#137]
hello, 
it's already the 4th time i have to resync the blockchain from beginning since it seems to save properly but resulting in the file corrupted
.
log below  1am - 11am

2018-Jan-04 01:11:16.673685 [SRV_MAIN]net_service loop stopped.
2018-Jan-04 01:11:16.923316 [SRV_MAIN]p2p net loop stopped
2018-Jan-04 01:11:17.115301 [SRV_MAIN]Stopping core rpc server...
2018-Jan-04 01:11:17.332852 [SRV_MAIN]Deinitializing core...
2018-Jan-04 01:11:17.560426 [SRV_MAIN]Mining has been stopped, 0 finished
2018-Jan-04 01:11:17.570953 [SRV_MAIN]Storing blockchain...
2018-Jan-04 01:12:16.656727 [SRV_MAIN]Blockchain stored OK.
2018-Jan-04 01:12:16.661239 [SRV_MAIN]Deinitializing rpc server ...
2018-Jan-04 01:12:16.662742 [SRV_MAIN]Deinitializing cryptonote_protocol...
2018-Jan-04 01:12:16.664246 [SRV_MAIN]Deinitializing p2p...
2018-Jan-04 01:12:16.676778 [SRV_MAIN]Node stopped.
2018-Jan-04 01:12:16.684798 [SRV_MAIN]Mining has been stopped, 0 finished
2018-Jan-04 11:09:03.803752 aeon v0.9.14.0()
2018-Jan-04 11:09:03.804754 Module folder: C:\aeon\aeond.exe
2018-Jan-04 11:09:03.843352 Initializing p2p server...
2018-Jan-04 11:09:03.854380 Binding on 0.0.0.0:11180
2018-Jan-04 11:09:03.855382 Net service binded on 0.0.0.0:11180
2018-Jan-04 11:09:03.856886 Attempting to add IGD port mapping.
2018-Jan-04 11:09:06.163213 Added IGD port mapping.
2018-Jan-04 11:09:06.164215 P2p server initialized OK
2018-Jan-04 11:09:06.165218 Initializing cryptonote protocol...
2018-Jan-04 11:09:06.166220 Cryptonote protocol initialized OK
2018-Jan-04 11:09:06.167223 Initializing core rpc server...
2018-Jan-04 11:09:06.168225 Binding on 127.0.0.1:11181
2018-Jan-04 11:09:06.169228 Core rpc server initialized OK on port: 11181
2018-Jan-04 11:09:06.170732 Initializing core...
2018-Jan-04 11:09:06.181258 Loading blockchain...
2018-Jan-04 11:09:16.228141 ERROR C:\Users\Arno\Documents\GitHub\aeon-0.9.14.0\src\common/boost_serialization_helper.h:108 Exception at [unserialize_obj_from_file], what=bad allocation
2018-Jan-04 11:09:16.997084 Can't load blockchain storage from file, generating genesis block.
2018-Jan-04 11:09:17.404613 ERROR C:\Users\Arno\Documents\GitHub\aeon-0.9.14.0\src\cryptonote_core\blockchain_storage.cpp:768 Block with id: c4890ac26e2806f95859c7725d66a248643fbc6b8fefcb5b8f408c24aa3c6f56 (as alternative) have wrong miner transaction
2018-Jan-04 11:09:19.201152 ERROR C:\Users\Arno\Documents\GitHub\aeon-0.9.14.0\src\cryptonote_core\blockchain_storage.cpp:130 Failed to add genesis block to blockchain
2018-Jan-04 11:09:19.778110 ERROR C:\Users\Arno\Documents\GitHub\aeon-0.9.14.0\src\cryptonote_core\cryptonote_core.cpp:128 Failed to initialize blockchain storage
2018-Jan-04 11:09:20.215716 ERROR C:\Users\Arno\Documents\GitHub\aeon-0.9.14.0\src\daemon\daemon.cpp:191 Failed to initialize core
2018-Jan-04 11:09:20.597179 Mining has been stopped, 0 finished

**Comment by stoffu** [comment]
Looks quite strange, seems like the daemon generates the genesis block but the generated block is considered invalid. Have you tried deleting the Aeon blockchain folder (C:/Users/<username>/AppData/Roaming/aeon)?

**Comment by mega2k0** [comment]
yes, the only way to fix this is to delete blockchain.bin (but its the 4th time,and i don't understand why it corrupt the files every single time) and redownload 5GB 
to "refresh" blockchain in simplewallet i have to use any public aeon nodes tho (working fine) 
TLDR
exiting or saving in aeod just corrupt the file everytime

**Comment by stoffu** [comment]
That's very strange. What if you did the following?

- Delete the aeon blockchain folder
- Run aeond.exe and let it download the first few hundred blocks, then type `save` and `exit`
- Run aeond.exe again

Will this result in the same corruption error? This test should take only a few minutes.

**Comment by mega2k0** [comment]
i actually did someting else:
i deleted the aeon's folder in Roaming
restored the old blockchain.bin (5gb)
making the daemon recreate pspstate and poolstate
its now working again!
.so the problem was not a corrupted blockchain.bin but the other two files (p2pstate and poolstate)
result

2018-Jan-04 14:18:43.071137 [P2P2][78.46.139.6:11180 OUT] SYNCHRONIZED OK
2018-Jan-04 14:18:43.785943 [P2P1][5.9.60.176:11180 OUT] SYNCHRONIZED OK
2018-Jan-04 14:18:43.787447 [P2P2][95.213.244.245:11180 OUT] SYNCHRONIZED OK
2018-Jan-04 14:18:51.588656 [P2P2][195.208.54.112:11180 OUT] SYNCHRONIZED OK

**Comment by aeonix** [comment]
Not fully explained but unable to reproduce and appears resolved
